vault/data_source_azure_access_credentials.go

```diff
@@ -125,87 +128,84 @@ func azureAccessCredentialsDataSourceRead(d *schema.ResourceData, meta interface
 		return nil
 	}
 
-	secret, err = client.Logical().Read(configPath)
+	config, err := client.Logical().Read(configPath)
 	if err != nil {
 		return fmt.Errorf("error reading from Vault: %s", err)
 	}
-	log.Printf("[DEBUG] Read %q from Vault", configPath)
+	log.Printf("[DEBUG] Successfully read %q from Vault", configPath)
 
 	subscriptionID := ""
```

Comment:
Instead of leveraging the subscriptionId and tenantId returned from the Vault configuration, can these be optionally defined in the data source? Otherwise it'll require all Azure SPs in Vault to be granted access to a centralized Subscription that is defined in the Vault configuration, which is likely not needed by the SP that's generated for anything else.

Reply:
Sure, we can add that, but I think that it is out of scope for this PR. I can open another PR, but it probably won't make it into 3.4.0. Sound okay?

Reply:
I'm concerned it'll cause the provider to break for any existing vault roles that don't have access to the subscription that's defined in the Azure Secrets engine config in Vault.

Reply:
PR forthcoming

Reply:
Addressed in #1384